src/Compilers/CSharp/Portable/BoundTree/BoundTreeVisitors.cs

```diff
@@ -184,22 +184,23 @@ public static Location GetTooLongOrComplexExpressionErrorLocation(BoundNode node
             {
                 SyntaxNode syntax = node.Syntax;
 
-                if (!(syntax is ExpressionSyntax))
+                if (syntax is not (ExpressionSyntax or PatternSyntax))
```

Comment:
>(ExpressionSyntax or PatternSyntax) [](http://example.com/codeflow?start=34&length=35)

Consider creating a static local function since this same check is used three times.

Reply:
This doesn't really feel like the type of thing that needs to be extracted to a local function; it honestly feels like that would impede readability, not help. I'd prefer to leave this as is.